lisa/sut_orchestrator/azure/common.py

```diff
@@ -195,24 +210,209 @@ class AzureVmPurchasePlanSchema:
 
 @dataclass_json
 @dataclass
-class AzureImageSchema:
+class AzureImageSchema(schema.ImageSchema):
+    architecture: Union[
+        schema.ArchitectureType, search_space.SetSpace[schema.ArchitectureType]
+    ] = field(  # type: ignore
+        default_factory=partial(
+            search_space.SetSpace,
+            is_allow_set=True,
+            items=[schema.ArchitectureType.x64, schema.ArchitectureType.Arm64],
+        ),
+        metadata=field_metadata(
+            decoder=partial(
+                search_space.decode_nullable_set_space,
+                base_type=schema.ArchitectureType,
+                is_allow_set=True,
+                default_values=[
+                    schema.ArchitectureType.x64,
+                    schema.ArchitectureType.Arm64,
+                ],
+            )
+        ),
+    )
+    disk_controller_type: Optional[
+        Union[
+            search_space.SetSpace[schema.DiskControllerType], schema.DiskControllerType
+        ]
+    ] = field(  # type:ignore
+        default_factory=partial(
+            search_space.SetSpace,
+            is_allow_set=True,
+            items=[schema.DiskControllerType.SCSI, schema.DiskControllerType.NVME],
+        ),
+        metadata=field_metadata(
+            decoder=partial(
+                search_space.decode_nullable_set_space,
+                base_type=schema.DiskControllerType,
+                is_allow_set=True,
+                default_values=[
+                    schema.DiskControllerType.SCSI,
+                    schema.DiskControllerType.NVME,
+                ],
+            )
+        ),
+    )
+    hyperv_generation: Optional[
```

Comment:
Other places are using name `hyper_v_generation`, but here is `hyperv_generation`. Should they are the same?

Reply:
Images are tagged with `hyper_v_generation` naming convention in LISA has been `hyperv_generation`